packages/integration-tests/src/entities/User.ts

```diff
@@ -1,7 +1,8 @@
-import { UserCodegen } from "./entities";
-
-import { userConfig as config } from "./entities";
-
+import { UserCodegen, userConfig as config } from "./entities";
```

Comment:
I'm surprised there isn't a new line after this `import`... looks like maybe Babel is dropping all new lines when it does `generate`?

I wonder if there is a way to avoid that b/c someone Jest's `toMatchInlineSnapshot` does not, although maybe the difference is that they use prettier which just puts the lines back, and my guess is that dprint probably removes extra new lines, but isn't as opinionated about adding newlines if necessary... Wonder if that's an option...

Reply:
I had another look at Jest, they used a generate option I hadn't enabled `{ retainLines: true }` but alas this still doesn't do the job. So I'm curious at how they are doing that - perhaps it is prettier.

From what I've seen, the Babel AST doesn't retain whitespace, so this isn't going to work how we want it to.

The option here is to use https://github.com/benjamn/recast which keeps track of the original ast to only print the bits that have changed.

Reply:
Just a naive question, but per "retainLines retains too much", is that how in the screenshot the `/** Case is preserved...` is on the same line as `name: string`? Or what is the unexpected retainage?

For the `/** Case is preserved`, if we just add our own newline ...or equivalent AST entry...? would that nudge it down to the next line?

Reply:
Correct. It seems like retainLines is largely there to not disrupt line numbers, stack traces etc.

Babel docs:
> This option exists so that users who cannot use source maps can get vaguely useful error line numbers.

I'm thinking it's worth perhaps biting the bullet here and trying out recast, because if it works as I think it should, it will completely absolves us of needing to do any reformatting on user files. and skips the prettier/dprint stuff.